internal/tofu/node_resource_plan_instance.go

```diff
@@ -554,6 +554,17 @@ func (n *NodePlannableResourceInstance) importState(ctx EvalContext, addr addrs.
 		return instanceRefreshState, diags
 	}
 
+	// insert marks from schema
+	if n.Config != nil {
+		valueWithSchemaMarks, _, configDiags := ctx.EvaluateBlock(n.Config.Config, n.Schema, nil, EvalDataForNoInstanceKey)
+		diags = diags.Append(configDiags)
+		if configDiags.HasErrors() {
+			return instanceRefreshState, diags
+		}
+		_, pvm := valueWithSchemaMarks.UnmarkDeepWithPaths()
+		instanceRefreshState.Value = instanceRefreshState.Value.MarkWithPaths(pvm)
+	}
```

Comment:
This fix is specific to the import block, and does not affect the `tofu import` command, which have the same results.

Also, feels a bit weird to me that we need to perform HCL block evaluation here in order to get the schema marks. You could probably simply rely on

Reply:
Thanks @RLRabinowitz,  actually, the refresh state contains schema marks defined in provider.
But in this case, the sensitive mark is from configuration(sensitive datasouce), so here I do a evaluation to avoid the difference between planned val and current state. And I edited my PR description to make it clear.

Reply:
As for the `tofu import`, I will continue work on it

Reply:
@chenzejun sorry if I wasn't clear here, but the following does not work with the current code, and should behave the same:

Let's say you have the following configuration fresh (same as with the issue, just without the import block):
```hcl
data "kubernetes_namespace" "ns" {
  metadata {
    name = "namespace-test"
  }
}

data "kubernetes_secret" "src" {
  metadata {
    name      = "secret-src"
    namespace = data.kubernetes_namespace.ns.id
  }
}

resource "kubernetes_secret" "dst" {
  metadata {
    name      = "secret-dst"
    namespace = data.kubernetes_namespace.ns.id
  }
  wait_for_service_account_token = false
  type                           = "kubernetes.io/basic-auth"
  data = data.kubernetes_secret.src.data
}
```

And then you:
1. `tofu init`
2. `tofu import "kubernetes_secret.dst" "namespace-test/secret-dst"`
3. `tofu plan`

You would get the same result of an unnecessary "will-update-in-place" plan

Reply:
Hi, @RLRabinowitz I have also fixed the import command, PTAL again